data/tutorials/getting-started/gs_02_fst_program.md

```diff
@@ -0,0 +1,343 @@
+
+```shell
+$ cat ./bin/main.ml
+let () = print_endline "Hello, World!"
+```
+
+The project-wide metadata is available in the `dune-project` file. Each folder containing source files that need to be built must contain a `dune` file explaining how.
+
+This builds the project:
+```shell
+$ dune build
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+```
+
+This launches the executable it creates:
+```shell
+$ dune exec hello
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+Hello, World!
+```
+
+Let's see what happens when we edit the `bin/main.ml` file directly. Navigate into it and replace the world `World` with your first name. Recompile the project with `dune build` as before, and then launch it again with `dune exec hello`.
+
+Voilà! You've just written your first OCaml program.
+
+In the rest of this tutorial, we will make more changes to this project in order to illustrate OCaml's tooling.
+
+## Why Isn't There a Main Function?
+
+Although `bin/main.ml`'s name accurately suggests it contains the application entry point into the executable binary produced by the project, it does not contain a dedicated `main` function. Even that file name is irrelevant, it could be anything. This is because OCaml is equally an interpreted and compiled language. The interpreter reads input and executes it, triggering computation and side effects. Execution of the binaries generated by the compiler are behaves as if interpretation was taking place, the only difference should be speed.
+
+That's why double semicolumns aren't needed in source files like they are in the toplevel. Statements are just processed in order from top to bottom, each triggering the side effects it may have. Definitions are added to the environment. Values resulting from nameless expressions are ignored. Side effects from all those will take place in the same order. That's OCaml main.
+
+However, it is common practice to single out a value that triggers all the side effects and mark it as the intended main entry point. In OCaml, that's the role of `let () =`, which stands for “definition which doesn't create a name” and actually indicates we're only interested in the side effects of the value on the right.
+
+## Modules and the Standard Library, Cont'd
+
+Let's summarise what was said about modules in the previous tutorial:
+- Modules are bundles of named values
+- Identical names from distinct modules don't clash
+- The standard library is a bunch of modules
+
+In addition to that, modules allows organising projects. Concerns are separated into isolated modules. This is outlined in the next section. Before creating a module, here is how to use a definition from a module in our project.
+```shell
+$ echo 'let () = Printf.printf "%s!\\n" "Hello, World!"' > bin/main.ml
```

Comment:
I know that the tutorial mentions that the expected level of people who go through this tutorial should be at least at some level of familiarity with programming but I still feel like using terminal commands like `echo` for changing files might not be the most beginner-friendly way. Especially for people who don't rely on CLI workflows a lot.

What do you think about changing these examples by actually writing the OCaml code?

I feel like it's as easy to copy-paste this code from here to your own file as to copy-paste this `echo` command to your terminal and run it.

You'll also get some nice syntax highlighting in your examples as a bonus 🙂 

```ocaml
(* contents of: bin/main.ml *)
let () = Printf.printf "%s!\\n" "Hello, World!"
```